File: Examples.VB/Features/Charts/NewCharts/AddFunnelChart.vb
Namespace Features.Charts.NewCharts
    Public Class AddFunnelChart
        Inherits ExampleBase

        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim worksheet As IWorksheet = workbook.Worksheets(0)

            worksheet.Range("A1:B7").Value = New Object(,) {
                {"Stage", "Amount"},
                {"Prospects", 500},
                {"Qualified prospects", 425},
                {"Needs analysis", 200},
                {"Price quotes", 150},
                {"Negotiations", 100},
                {"Closed sales", 90}
            }
            worksheet.Range("A:A").Columns.AutoFit()

            'Create a funnel chart.
            Dim shape As IShape = worksheet.Shapes.AddChart(ChartType.Funnel, 300, 20, 300, 200)
            shape.Chart.SeriesCollection.Add(worksheet.Range("A1:B7"))

            'Set the axis invisible.
            Dim axis As IAxis = shape.Chart.Axes.Item(AxisType.Category, AxisGroup.Primary)
            axis.Visible = False
        End Sub
        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property ShowScreenshot As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides ReadOnly Property IsNew As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace

File: Examples/Features/Charts/NewCharts/AddFunnelChart.cs
using GrapeCity.Documents.Excel.Drawing;
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Charts.NewCharts
{
    public class AddFunnelChart : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            IWorksheet worksheet = workbook.Worksheets[0];

            worksheet.Range["A1:B7"].Value = new object[,]
            {
                {"Stage", "Amount"},
                {"Prospects", 500},
                {"Qualified prospects", 425},
                {"Needs analysis", 200},
                {"Price quotes", 150},
                {"Negotiations", 100},
                {"Closed sales", 90}
            };
            worksheet.Range["A:A"].Columns.AutoFit();

            //Create a funnel chart.
            IShape shape = worksheet.Shapes.AddChart(ChartType.Funnel, 300, 20, 300, 200);
            shape.Chart.SeriesCollection.Add(worksheet.Range["A1:B7"]);

            //Set the axis invisible.
            IAxis axis = shape.Chart.Axes.Item(AxisType.Category, AxisGroup.Primary);
            axis.Visible = false;            
        }
        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }

        public override bool ShowScreenshot
        {
            get
            {
                return true;
            }
        }
    }
}
